tx-pool/src/process.rs

```diff
@@ -202,13 +217,34 @@ impl TxPoolService {
             .with_tx_pool_read_lock(|tx_pool, snapshot| {
                 let tip_hash = snapshot.tip_hash();
 
+                // Same txid means exactly the same transaction, including inputs, outputs, witnesses, etc.
+                // It's also not possible for RBF, reject it directly
                 check_txid_collision(tx_pool, tx)?;
 
-                let (rtx, status) = resolve_tx(tx_pool, &snapshot, tx.clone())?;
-
-                let fee = check_tx_fee(tx_pool, &snapshot, &rtx, tx_size)?;
-
-                Ok((tip_hash, rtx, status, fee, tx_size))
+                // Try normal path first, if double-spending check success we don't need RBF check
+                // this make sure RBF won't introduce extra performance cost for hot path
+                let res = resolve_tx(tx_pool, &snapshot, tx.clone(), false);
+                match res {
+                    Ok((rtx, status)) => {
+                        let fee = check_tx_fee(tx_pool, &snapshot, &rtx, tx_size)?;
+                        Ok((tip_hash, rtx, status, fee, tx_size, HashSet::new()))
+                    }
+                    Err(err) => {
+                        if tx_pool.enable_rbf() && matches!(err, Reject::Resolve(_)) {
```

Comment:
Is it possible to have a scenario where `err` is `RBFRejected` when we reach this line?

Reply:
No possible, the `resolve_tx_from_pool` always return `Reject::Resolve` for error.